runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1285,16 +1385,62 @@ TR::Node *TR_VectorAPIExpansion::storeIntrinsicHandler(TR_VectorAPIExpansion *op
       }
    else if (mode == checkVectorization)
       {
-      if (objType != Vector)  // TODO: Mask and Shuffle
-         return NULL;
+      if (objType == Vector)
+         {
+         TR::DataType vectorType = TR::DataType::createVectorType(elementType, vectorLength);
+         TR::ILOpCodes vectorOpCode = TR::ILOpCode::createVectorOpCode(TR::vstorei, vectorType);
 
-      TR::DataType vectorType = TR::DataType::createVectorType(elementType, vectorLength);
-      TR::ILOpCodes vectorOpCode = TR::ILOpCode::createVectorOpCode(TR::vstorei, vectorType);
+         if (!comp->cg()->getSupportsOpCodeForAutoSIMD(vectorOpCode))
+            return NULL;
 
-      if (!comp->cg()->getSupportsOpCodeForAutoSIMD(vectorOpCode))
-         return NULL;
+         return node;
+         }
+      if (objType == Mask)
```

Comment:
Please change this line to `else if (objType == Mask)`.
I think that is clearer, and consistent with other lines such as L.1192 and L.1308.